hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+      desc: '''Error-related interrupts, see ERROR_MASK register for more
+               information.'''
+    },
+    { name: "event",
+      desc: '''Event-related interrupts, see EVENT_MASK register for more
+               information.'''
+    }
+  ],
+  inter_signal_list: [
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sck_i",
+      act:     "rcv",
+      width:   1,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_cs_ni",
+      act:     "rcv",
+      width:   8,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdo_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdt_i",
+      act:     "rcv",
+      width:   4,
+      package: ""
+    },
+    { struct:  "logic",
+      type:    "unit",
+      name:    "pt_sdi_o",
+      act:     "req",
+      width:   4,
+      package: ""
+    },
+  ]
+  registers: [
+    { name: "CONTROL",
+      desc: "Control register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      fields: [
+        { bits: "31",
+          name: "SPIEN",
+          desc: '''Enables the SPI host.  On reset, this field is 0, meaning that no transactions can proceed.'''
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "RST_FSM",
+          desc: '''This bit holds the control FSM in reset when set to 1.'''
+          resval: "0x0",
+          hwqe: "true",
+          hwacces: "hrw"
+        },
+        { bits: "29",
+          name: "RST_TXFIFO",
+          desc: '''Resets the TX FIFO when set to 1.'''
+          resval: "0x0",
+        },
+        { bits: "28",
+          name: "RST_RXFIFO",
+          desc: '''Resets pulse the RX FIFO when set to 1.'''
+          resval: "0x0"
+        },
+        { bits: "27",
+          name: "PASSTHRU",
+          desc: '''Enables Pass-through mode, wherein the SPI_HOST IP surrenders
+                   control of the SPI bus to another block. In this mode the
+                   `sck`, `cs_n` and `sd` signals are no longer driven by
+                   the SPI_HOST IP, rather they are multiplexed out to match the
+                   on-chip inputs `pt_sck_i`, `pt_cs_ni`, `pt_sdt_i`, `pt_sdo_i`,
+                   and `pt_sdi_o`.  (Since the `sd` bus is bidirectional, there are
+                   separate input, output and tri-state controls for this bus.)''',
+          resval: "0x0"
+        },
+        { bits: "24:16",
+          name: "TX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
+                   depth of the TX FIFO is ever less than
+                   TX_WATERMARK words (32b each). Set to zero
+                   to temporarily disable interrupts.'''
+          resval: "0"
+        },
+        { bits: "15:9",
+          name: "RX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
```

Comment:
Maybe it's worth explicitly saying when this happens, with a link. Something like this, maybe?
```
        { bits: "15:9",
          name: "RX_WATERMARK"
          desc: '''If !!EVENT_ENABLE.RXWM is set, the IP will send
                   an interrupt if the depth of the RX FIFO is ever greater than
                   RX_WATERMARK words (32b each).'''
          resval: "127"
        },
```

And similarly for TX_WATERMARK.

Reply:
I like that! Thank you.
Here are my minor tweaks:

        { bits: "24:16",
          name: "TX_WATERMARK"
          desc: '''If !!EVENT_ENABLE.TXWM is set, the IP will send
                   an interrupt when the depth of the RXFIFO drops below
                   TX_WATERMARK words (32b each).'''
          resval: "0"
        },
        { bits: "15:9",
          name: "RX_WATERMARK"
          desc: '''If !!EVENT_ENABLE.RXWM is set, the IP will send
                   an interrupt when the depth of the RXFIFO reaches
                   RX_WATERMARK words (32b each).'''
          resval: "127"
        },